andela/knights-bn-backend
# #170947667 Notifications for new travel request

#### What does this PR do?

- This PR will notify the user that their request was sent

- This PR will notify the manager that a new request was made

#### Description of Task to be completed?

- create an end point to notify the manager that a new request was made

- Test that endpoint works

#### How should this be manually tested?
- run `git clone https://github.com/andela/knights-bn-backend.git` in your terminal
- run `cd knights-bn-backend` in your terminal
- run `git checkout ft-new-request-notification-170947667` to go on the branch
- run `npm install` to install all packages. 
- create `.env` file and populate it, refer to `env.example` to see what fill in .env
- create `database` for both `development` and `test`
- fill the names of the databases you created under `.env`
-  run `npm run createTables` to create tables
- run `npm run start` to start the server.
- use `postman` or other API tester

1. signup a new user (or login if that user already exits)
           -  POST `localhost:4000/api/v1/auth/signup`
                           request body example
            
                 {
                        "firstName": "newiyonsenga",
                         "lastName": "Eric",
                         "gender": "male",
                         "passportNumber": "1234beug",
                        "email": "eugen@gmail.com",
                        "password": "Niyonkuru@1"
                   }

2. create Request 
          - POST `http://localhost:4000/api/v1/trips/returnTrip`
                            request body example

                    {
	                   "origin": "Kigali",
	                  "destination": "Kampala",
	                  "departureDate": "2024-03-28",
	                  "returnDate": "2024-04-28",
	                  "accommodation": "XYZ campus",
	                  "reason": "Having fun",
	                 "passportNumber": "12345677"
                     } 

#### Any background context you want to provide?

- N/A

#### What are the relevant pivotal tracker stories?

- [#170947667](https://www.pivotaltracker.com/story/show/170947667)

#### Screenshots

     An email users get when they make a request

<img width="468" alt="Screen Shot 2020-02-25 at 10 02 48" src="https://user-images.githubusercontent.com/48021034/75227042-208d0a80-57b6-11ea-8e54-5a12c9320eda.png">
